src/D2L.CodeStyle.Analyzers/Immutability/ImmutableDefinitionChecker.cs

```diff
@@ -53,27 +53,34 @@ ImmutabilityContext context
 			}
 
 			foreach( var member in members ) {
-				if( IsAudited( member ) ) {
-					if( CheckMember( diagnosticSink: _ => { }, member ) ) {
-						m_diagnosticSink(
-							Diagnostic.Create(
-								Diagnostics.UnnecessaryMutabilityAnnotation,
-								GetLocationOfAnnotation( member )
-							)
-						);
-					}
-
-					continue;
-				}
-
-				if( !CheckMember( m_diagnosticSink, member ) ) {
+				if( !CheckMember( member ) ) {
 					result = false;
 				}
 			}
 
 			return result;
 		}
 
+		public bool CheckMember( ISymbol member ) {

```

Comment:
This method is public. It will also be called to check static fields/properties (for any class/struct, not just `[Immutable]` ones which would otherwise go through the analyzer).

It calls into the private `CheckMember` which lets you sub out the diagnostic sink.